actors/builtin/verifreg/verified_registry_actor.go

```diff
@@ -228,12 +265,29 @@ func (a Actor) RestoreBytes(rt vmr.Runtime, params *RestoreBytesParams) *adt.Emp
 	}
 
 	var st State
+	rt.State().Readonly(&st)
+	// TODO We need to resolve the client address to an ID address before making this comparison.
+	// https://github.com/filecoin-project/specs-actors/issues/556
+	if st.RootKey == params.Address {
+		rt.Abortf(exitcode.ErrIllegalArgument, "Cannot restore allowance for Rootkey")
+	}
+
 	rt.State().Transaction(&st, func() interface{} {
 		verifiedClients, err := adt.AsMap(adt.AsStore(rt), st.VerifiedClients)
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load verified clients")
 
+		verifiers, err := adt.AsMap(adt.AsStore(rt), st.Verifiers)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load verifiers")
+
+		// validate we are NOT attempting to do this for a verifier
+		found, err := verifiers.Get(AddrKey(params.Address), nil)
```

Comment:
This check seems expensive (loads a HAMT) and pointless. Are we expecting this case to be possible @aarshkshah1992?